### PR TITLE
Type annotations for `mlx.core` module

### PR DESCRIPTION
**Pull request by danilopeixoto**
## Proposed changes

Distribute `mlx.core` stubs with package to provide type annotations for module.
#19

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -121,10 +133,30 @@ jobs:
...
       - run:
-          name: Build package
+          name: Build python package
+          command: |
+            eval "$(conda shell.bash hook)"
+            conda activate runner-env
+            DEVELOPER_DIR=$(developer_dir_macos_<< parameters.macos_version >>) \
+              PYPI_RELEASE=1 \
+              CMAKE_BUILD_PARALLEL_LEVEL="" \
+              python setup.py build_ext
```
Comment (angeloskath) [inline .circleci/config.yml:147]: I don't think there is the need for `build_ext` before `install`. Install will take care of building anyway.

File: .circleci/config.yml [hunk 2]
```diff
@@ -157,10 +189,30 @@ jobs:
...
       - run:
-          name: Build package
+          name: Build python package
+          command: |
+            eval "$(conda shell.bash hook)"
+            conda activate runner-env
+            DEVELOPER_DIR=$(developer_dir_macos_<< parameters.macos_version >>) \
+              DEV_RELEASE=1 \
+              CMAKE_BUILD_PARALLEL_LEVEL="" \
+              python setup.py build_ext
```
Comment (angeloskath) [inline .circleci/config.yml:203]: Same as above.

File: .circleci/config.yml [hunk 3]
```diff
@@ -193,10 +245,28 @@ jobs:
...
             pip install twine
       - run:
-          name: Build package
+          name: Build python package
+          command: |
+            eval "$(conda shell.bash hook)"
+            conda activate runner-env
+            DEVELOPER_DIR=$(developer_dir_macos_<< parameters.macos_version >>) \
+              CMAKE_BUILD_PARALLEL_LEVEL="" \
+              python setup.py build_ext
```
Comment (angeloskath) [inline .circleci/config.yml:258]: And last one :-) .

**Review by angeloskath — CHANGES_REQUESTED** [review]
This looks great, thanks!

I left one minor comment, basically we can remove the build before install.

File: .circleci/config.yml (hunk 1 above)
Comment (danilopeixoto) [reply .circleci/config.yml:147]: Yes, thank you for noticing that. It's not necessary for builds that are not in-place. I'm fixing that.

File: .circleci/config.yml (hunk 2 above)
Comment (danilopeixoto) [reply .circleci/config.yml]: Fixed.

File: .circleci/config.yml (hunk 3 above)
Comment (danilopeixoto) [reply .circleci/config.yml]: Fixed.

**Review by angeloskath — APPROVED** [review]
Thanks that looks good!

**Comment by danilopeixoto** [comment]
@angeloskath `python setup.py install` (deprecated) command is attempting to install dependencies based on remote name. We can use `python -m pip install .` instead.

**Pull request merged**